examples/dataflow-hashpipeline/hashpipeline.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# pytype: skip-file
+
+from __future__ import absolute_import
+
+import logging
+import sys
+import json
+from argparse import ArgumentTypeError
+
+import apache_beam as beam
+from apache_beam.transforms.window import FixedWindows
+from apache_beam.io import ReadFromPubSub
+from apache_beam.io import WriteToPubSub
+from apache_beam.options.pipeline_options import StandardOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+
+# NOTE: In the latest versions of Beam for Python there are several issues
+# wth saving the main session that cause bugs in dataflow, so instead of
+# requiring at the top of the file, we do so inside the DoFns and PTransforms
+# See: BEAM-10150 for more
+
+class HashPipelineOptions(PipelineOptions):
+  @classmethod
+  def _add_argparse_args(cls, parser):
+    parser.add_argument('--input_subscription',
+      help='Fully qualified input Pubsub subscription that forwards raw events from GCS. i.e. projects/*/subscriptions/*')
```

Comment:
Consider enforcing this with [argparse type regex checker](https://stackoverflow.com/questions/41881002/python-argparse-regex-expression)

Reply:
This is already checked by the IO class when the pipeline is being constructed

Reply:
good point.